force_wfmanager/wfmanager_task.py

```diff
@@ -54,13 +57,22 @@ def create_dock_panes(self):
 
     def save_workflow(self):
         """ Shows a dialog to save the workflow into a JSON file """
+        writer = WorkflowWriter()
+
+        # If the user already saved before or loaded a file, we overwrite this
+        # one
+        if len(self.current_file) != 0:
+            with open(self.current_file, 'wr') as output:
+                writer.write(self.workflow_m, output)
+            return
+
         dialog = FileDialog(action="save as", default_filename="workflow.json")
         result = dialog.open()
 
         if result is OK:
-            writer = WorkflowWriter()
             with open(dialog.path, 'wr') as output:
                 writer.write(self.workflow_m, output)
+            self.current_file = dialog.path
```

Comment:
Conglomerate the two operations, but don't store the file in the current_file until you are sure that the saving is successful.